rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1017,13 +1017,51 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
-    protected void processSVPFailure(Federation proposedFederation) {
+    protected void updateSvpState(Transaction rskTx) {
```

Comment:
To be honest, I'd find it easier to follow if the logic was written in chronological order. I know it requires more validations but we can probably make use of auxiliary methods. The way it's written now, for every new condition that appears in the code, you need to keep in mind the previous condition that was already validated a few lines above and the protocol logic about one having one SVP related value in storage at once.

I'd go with a very straight forward approach 

``` 
if (!proposedFederationExists) return;

if (validationPeriodIsOngoing) {
     if (shouldCreateSvpFundTx()) {
        createFundTx()
        return;
      }
     if (shouldCreateSvpSpendTx()) {
        createSpendTx()
        return;
      }

      validationFailed();
}
```
What do you think? I think that's the only 3 cases we need to cover

Reply:
makes sense, will update

Reply:
updated